PocketKit/Sources/Sync/PocketSource.swift

```diff
@@ -546,26 +546,28 @@ extension PocketSource {
         try? space.filterTags(with: input, excluding: tags)
     }
 
-    public func fetchDetails(for savedItem: SavedItem) async throws {
-        Log.breadcrumb(category: "sync", level: .debug, message: "Fetching detals for item with id \(String(describing: savedItem.remoteID))")
+    public func fetchDetails(for savedItem: SavedItem) async throws -> Bool {
+        Log.breadcrumb(category: "sync", level: .debug, message: "Fetching details for item with id \(String(describing: savedItem.remoteID))")
 
         guard let remoteID = savedItem.remoteID else {
-            return
+            return false
         }
 
         guard let remoteSavedItem = try await apollo
             .fetch(query: SavedItemByIDQuery(id: remoteID))
             .data?.user?.savedItemById else {
-            return
+            return false
         }
 
-        try space.performAndWait {
+        return try space.performAndWait {
             guard let savedItem = space.backgroundObject(with: savedItem.objectID) as? SavedItem else {
                 Log.capture(message: "Could not retreive item from background context for mutation")
-                return
+                return false
             }
             savedItem.update(from: remoteSavedItem.fragments.savedItemParts, with: space)
             try space.save()
+
+            return remoteSavedItem.item.asItem?.marticle?.isEmpty == false
```

Comment:
discussed with roux, curious to see if there was a better way to make it so we can use async / await by waiting for core data to return the status of successfully saving an item, we would not have to return this boolean and make that synchronous instead. It seems like there isn't a way, but opening up for the team in case there is.